src/foam/nanos/dig/HTTPDigestSink.java

```diff
@@ -73,9 +75,57 @@ public HTTPDigestSink(String url, String bearerToken, DUGDigestConfig dugDigestC
 
   @Override
   public void put(Object obj, Detachable sub) {
+    try {
+      int responseCode = sendRequest(obj);
+
+      if ( responseCode == HttpServletResponse.SC_OK ) return;
+
+      if ( responseCode == HttpServletResponse.SC_BAD_REQUEST ) { // client error
+        if ( obj instanceof EmailWebhook ) {
+          EmailWebhook ewh = (EmailWebhook) obj ;
+          DAO alarmDAO = (DAO) getX().get("alarmDAO");
+          String emailAddress = ewh.getEmail();
+          String note = "[" + emailAddress + ", " + new Date() + "]";
+          String name = "EmailWebhook";
+          Alarm alarm = (Alarm) alarmDAO.find(EQ(Alarm.NAME, name));
+
+          if ( alarm == null ) {
+            alarm = new Alarm.Builder(getX())
+              .setName(name)
+              .setSeverity(LogLevel.ERROR)
+              .setNote("Email not sent to:\n\t" + note)
+              .build();
+          } else {
+            alarm.setNote(alarm.getNote() + "\n\t" + note);
+          }
+          alarmDAO.put(alarm);
+        }
+      } else if ( responseCode == HttpServletResponse.SC_INTERNAL_SERVER_ERROR )  { // server error
+        // make a new request
+        responseCode = sendRequest(obj);
+
+        if ( responseCode == HttpServletResponse.SC_OK ) return;
+
+        if ( responseCode == HttpServletResponse.SC_INTERNAL_SERVER_ERROR ) {
+          // create an alarm
+          Alarm alarm = new Alarm.Builder(getX())
+            .setName("HTTP POST Request Failed")
+            .setNote("Failed with 500 status code: [" + obj.getClass().getSimpleName() + "]")
+            .build();
+          ((DAO) getX().get("alarmDAO")).put(alarm);
+        }
+      }
+
+      throw new RuntimeException(this.getClass().getSimpleName() + ", HTTP POST request failed with " + responseCode);
```

Comment:
We should handle timeouts also either connection timeout or gateway timeout (504), we should not retry on timeout as state on remote side is unknown but we should create alarm